examples/hyperspectral_image_classification_with_gsmote.py

```diff
@@ -0,0 +1,176 @@
+from sklearn.model_selection import GridSearchCV
+from sklearn.model_selection import train_test_split
+from sklearn.model_selection import StratifiedKFold
+from sklearn.metrics import classification_report
+from sklearn.metrics import confusion_matrix
+from sklearn import datasets 
+
+from imblearn.pipeline import Pipeline
+from gsmote import GeometricSMOTE
+
+
+print(__doc__)
+
+###############################################################################
+# Indian Pines Test Site 3
+###############################################################################
+
+###############################################################################
+# This hyperspectral data set has 220 spectral bands and 20 m spatial resolution. 
+# The imagery was collected on 12 June 1992 and represents a 2.9 by 2.9 km area 
+# in Tippecanoe County, Indiana, USA. The area is agricultural and eight classes 
+# as land-use types are presented: alfalfa, corn, grass, hay, oats, soybeans, 
+# trees, and wheat. The Indian Pines data set has been used for testing and 
+# comparing algorithms. The number of samples varies greatly among the classes, 
+# which is known as an imbalanced training set. Data are made available by 
+# Purdue University (https://engineering.purdue.edu/~biehl/MultiSpec/hyperspectral.html)
+
+
+
+###############################################################################
+# Dataset
+###############################################################################
+
+###############################################################################
+# This dataset provides the data in numpay arrays. Predictor and target 
+# variables are already split (X and y accordingly). Predictor data consists 
+# of 220 features. Target attributes are the land cover classes. Dataset has 
+# 9144 samples.  
+
+
+# Importing the datad
+X, y, feature_names, description = tuple(fetch_openml('Indian_pines').values())[:4]
+
+n_features = (X).shape[1]
+n_samples = (X).shape[0]
+ratio = n_samples/n_features
+print("n_features:'%s', n_samples: '%s', Ratio: '%s'" %
+      (n_features, n_samples, ratio))
+
+# Data split into train and test parts
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42, 
+                                                    test_size=0.33, shuffle=True)
+
+###############################################################################
+# Classification
+###############################################################################
+
+###############################################################################
+# Below we present a simple implementation of Logistic Regression Classifier (LR) 
+# with Principal Component Analysis (PCA) and G-SMOTE oversampler (combined 
+# by pipeline). As the dimentionality of the dataset is huge (220) the PCA 
+# analysis become important to reduce the time taken for classification. Also,
+# as the dataset is multiclass, we use OneVsRestClassifier in order to carry
```

Comment:
You dont need it. Logistic regression does it automatically and it has a parameter to control the multi-class strategy.